app/models/manageiq/providers/kubernetes/container_manager_mixin.rb

```diff
@@ -102,19 +102,52 @@ def default_authentication_type
     :bearer
   end
 
-  def scan_job_create(entity_class, entity_id)
+  def scan_job_create(entity)
+    check_policy_prevent(:request_containerimage_scan, :raw_scan_job_create, entity)
+  end
+
+  def raw_scan_job_create(entity)
     Job.create_job(
       "ManageIQ::Providers::Kubernetes::ContainerManager::Scanning::Job",
       :name            => "Container image analysis",
-      :target_class    => entity_class,
-      :target_id       => entity_id,
+      :target_class    => entity.class.name,
+      :target_id       => entity.id,
       :zone            => my_zone,
       :miq_server_host => MiqServer.my_server.hostname,
       :miq_server_guid => MiqServer.my_server.guid,
       :ems_id          => id,
     )
   end
 
```

Comment:
the functions below duplicate host and vm and need to find a better place to live